addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/device/DSMRPort.java

```diff
@@ -0,0 +1,498 @@
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.TooManyListenersException;
+
+import org.openhab.binding.dsmr.internal.device.DSMRDeviceConstants.DSMRPortEvent;
+import org.openhab.binding.dsmr.internal.device.cosem.CosemObject;
+import org.openhab.binding.dsmr.internal.device.p1telegram.P1TelegramListener;
+import org.openhab.binding.dsmr.internal.device.p1telegram.P1TelegramParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gnu.io.CommPort;
+import gnu.io.CommPortIdentifier;
+import gnu.io.NoSuchPortException;
+import gnu.io.PortInUseException;
+import gnu.io.SerialPort;
+import gnu.io.SerialPortEvent;
+import gnu.io.SerialPortEventListener;
+import gnu.io.UnsupportedCommOperationException;
+
+/**
+ * Class that implements the DSMR port functionality that comply to the Dutch Smart Meter Requirements.
+ * <p>
+ * This class will handle communication with the Serial Port itself and notify listeners about events and
+ * received P1 telegrams
+ * <p>
+ * The read method will claim OS resources if necessary. If the read method
+ * encounters unrecoverable problems it will automatically close itself
+ * <p>
+ * The close method can be called asynchronous and will release OS resources.
+ * </p>
+ * <code>
+ * An example DSMR telegram in accordance to IEC 62056-21 Mode D.<br>
+ * /ISk5\2MT382-1000<br>
+ * 0-0:96.1.1(4B384547303034303436333935353037)<br>
+ * 1-0:1.8.1(12345.678*kWh)<br>
+ * 1-0:1.8.2(12345.678*kWh)<br>
+ * 1-0:2.8.1(12345.678*kWh)<br>
+ * 1-0:2.8.2(12345.678*kWh)<br>
+ * 0-0:96.14.0(0002)<br>
+ * 1-0:1.7.0(001.19*kW)<br>
+ * 1-0:2.7.0(000.00*kW)<br>
+ * 0-0:17.0.0(016*A)<br>
+ * 0-0:96.3.10(1)<br>
+ * 0-0:96.13.1(303132333435363738)<br>
+ * 0-0:96.13.0(303132333435363738393A3B3C3D3E3F303132333435363738393A3B3C3D3E3F<br>
+ * 303132333435363738393A3B3C3D3E3F303132333435363738393A3B3C3D3E3F<br>
+ * 303132333435363738393A3B3C3D3E3F)<br>
+ * 0-1:96.1.0(3232323241424344313233343536373839)<br>
+ * 0-1:24.1.0(03)<br>
+ * 0-1:24.3.0(090212160000)(00)(60)(1)(0-1:24.2.1)(m3)<br>
+ * (00000.000)<br>
+ * 0-1:24.4.0(1)<br>
+ * !<br>
+ * </code>
+ *
+ * @author M. Volaart
+ * @since 2.1.0
+ */
+public class DSMRPort implements SerialPortEventListener, P1TelegramListener {
+    /* logger */
+    private final Logger logger = LoggerFactory.getLogger(DSMRPort.class);
+
+    /* private object variables */
+    private final String portName;
+
+    /* serial port resources */
+    private SerialPort serialPort;
+    private BufferedInputStream bis;
+    private byte[] buffer = new byte[1024]; // 1K
+
+    /* state variables */
+    private DSMRPortSettings portSettings;
+    private DSMRPortSettings fixedPortSettings; // Used if DSMR binding has a static port configuration
+    private boolean lenientMode = false;
+
+    private class DSMRPortStatus {
+        private boolean isOpen = false;
+        private boolean bi = false;
```

Comment:
I think you refer to bi, oe, fe, pe. In the JavaDoc of SerialPortEvent also these names are used. Even in UART tutorials these abbreviations are used. For this reason i would like to keep these abbreviations. But if you insist :)